src/preact/parse-props.js

```diff
@@ -221,14 +219,12 @@ function parsePropDefs(Ctor, props, propDefs, element, mediaQueryProps) {
   }
 
   for (const name in propDefs) {
-    const def = /** @type {!AmpElementPropDef} */ (propDefs[name]);
+    const def = /** @type {AmpElementProp} */ (propDefs[name]);
     devAssert(
-      !!def.attr +
-        !!def.attrs +
-        !!def.attrMatches +
-        !!def.selector +
-        !!def.passthrough +
-        !!def.passthroughNonEmpty <=
+      Number(!!def.attr && !!def.attrs && !!def.attrMatches) +
+        Number(!!def.selector) +
+        Number(!!def.passthrough) +
+        Number(!!def.passthroughNonEmpty) <=
```

Comment:
My take on this existing-clunky check
```suggestion
      [
        def.attr,
        def.attrs,
        def.attrMatches,
        def.selector,
        def.passthrough,
        def.passthroughNonEmpty
      ].filter(Boolean).length <=
```

Reply:
Lets kick this cleanup to a different PR. I agree its incredibly clunky as is. Your diff is slightly incorrect though b/c the first 3 need to be grouped.

Reply:
The first 3 aren't grouped in the original code though, so this isn't quite a no-op. If the grouping is necessary (and in that case you're fixing a bug here) maybe at least leave a comment indicating that the grouping is intentional and a pointer to more info?

Reply:
oh you're right! idk how I messed that up....changing to your formulation